Makefile.one

```diff
@@ -30,24 +31,77 @@ CMD_CAT ?= cat
 CMD_MD5 ?= md5sum
 CMD_OPA ?= opa # https://github.com/open-policy-agent/opa/releases/download/v0.33.1/opa_linux_amd64
 
-check_%:
-	@command -v $* >/dev/null || (echo "missing required tool $*" ; exit 1)
+.ONESHELL:
+.check_%:
+#
+	@command -v $* >/dev/null
+	if [ $$? -ne 0 ]; then
+		echo "missing required tool $*"
+		exit 1
+	else
+		touch $@ # avoid target rebuilds due to inexistent file
+	fi
 
+#
 # libs
+#
 
 LIB_ELF ?= libelf
 LIB_ZLIB ?= zlib
 
-checklib_%:
-	@$(CMD_PKGCONFIG) --silence-errors --validate $* 2>/dev/null || \
-		(echo "missing lib $*" ; exit 1)
+.ONESHELL:
+.checklib_%: \
+	| .check_$(CMD_PKGCONFIG)
+#
```

Comment:
I've chosen this formatting style so we can always split the logic: build target dependency VS build commands. The option of having each dependency on its own line is to remove burden from maintenance (patches, for example, would change single lines for adding or removing build dependencies, etc). This is a well known strategy for sustaining engineering.